Mad_Bash/Assets/Import/FightOrFlight/Scripts/ButtonPressContext.cs

```diff
@@ -36,11 +36,14 @@ public int TurnCount
     [Tooltip("How many sequences")]
     public int MaxTurns = 4;
 
-
     [Tooltip("Adjust how long before the next state will execute")]
     public float StateTransitionInterval = 1;
     private float stateTransitionInterval;
 
+    [SerializeField]
+    public GameEventArgs OnTimerStart;
```

Comment:
The naming of this timer should be specific to this context. OnContextTimerStart and OnContextTimerEnd. If you are going to raise an event then you should add comments on how it will be interpreted on the handler side. Ex: "This event is intended to be handled by casting the first argument as an IContext then using the CurrentState argument to derive what state it is currently in.

Reply:
bump

Reply:
@jeremybarzas